### DISCUSSION TITLE
No more than 10 inputs for reusable workflows?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#26781] [Actions]
<p>I am hitting what seems like an odd limit. I am only able to pass up to 10 inputs to a <code>workflow_call</code> triggered workflow, and I get an error related to the <code>workflow_dispatch</code> event.</p>
<blockquote>
<p><strong>Invalid workflow file</strong> : .path/to/workflow/file</p>
<p>handling usage of workflow “path/to/workflow/file”: workflow is invalid: you may only define up to 10 <code>inputs</code> for a <code>workflow_dispatch</code> event</p>
</blockquote>
<p>I can see why this limitation would make sense for <code>workflow_dispatch</code> since we might want to save space in the UI, but for <code>workflow_call</code> I think it does not - since even some of our actions have more than 10 inputs.</p>
<p>Trying to make a reusable workflow with some flexibility, with only 10 inputs seems like it would be difficult.</p>
<p>Is this actually by design? Or am I hitting a bug</p>

**Comment by jenschelkopf** [comment, marked as answer]
<p>Hi <a class="mention" href="/joshua-hancox_fedex">@joshua-hancox_fedex</a>, I’m the product manager for reusable workflows. You’ve run into a bug. The limit should be much higher. I’ve opened an internal issue for the team to investigate. Thanks for letting us know.</p>

**Comment by ghost** [comment]
<p>Cool, thankyou for the quick response. :slight_smile:</p>

**Comment by jenschelkopf** [comment]
<p>Hi <a class="mention" href="/joshua-hancox_fedex">@joshua-hancox_fedex</a>, we’ve removed the limit on the number of inputs that can be defined for a reusable workflow. See if that works for you!</p>

**Comment by ghost** [comment]
<p>Yes testing and working OK. Thankyou.</p>

**Comment by alexrudd** [comment]
@jenschelkopf

Sorry to post on an old discussion, but I just hit this same error message:

>Invalid workflow file: .github/workflows/release-branch.yml#L1]
you may only define up to 10 `inputs` for a `workflow_dispatch` event

Is the limit back?

**Reply by julian-smith-artifex-com** [reply]
I've just run into the same problem.

The drop-down list from the `Run workflow` button contained all the specified items, but the runner failed with:

```
you may only define up to 10 `inputs` for a `workflow_dispatch` event
```

**Reply by emma-roudabush-disney** [reply]
Same--any update on this?

**Reply by JamieStM** [reply]
FYI @ anyone that finds this: This question pertains to `workflow_call`, where one workflow calls another. This comment pertains to `workflow_dispatch`, where a user manually triggers a workflow in the UI. See discussion here for further updates: #8774. Notably see the relatively recent (Oct 2024) comment indicating near-term improvements [here](https://github.com/orgs/community/discussions/8774#discussioncomment-10890000).

**Comment by EvgenyMettsler** [comment]
@jenschelkopf

I've just run into the same issue:
```
GitHub Actions
Invalid workflow file
you may only define up to 10 `inputs` for a `workflow_dispatch` event
```
Any updates on that? We need at least 15-20 inputs for our deployment.

**Comment by EvgenyMettsler** [comment]
@jenschelkopf Would you mind to pass the issue to the team with a high priority please?

**Comment by isaachernandezMT** [comment]
Any update on this...?

**Comment by mandeepsingh-dh** [comment]
@jenschelkopf  Any updates on this? We need more than 10 inputs for our manually triggered jobs.

**Reply by jfrank-nih** [reply]
You're looking for #8774. This one is `workflow_call`, not `workflow_dispatch`.

**Comment by AbdulAhad365** [comment]
The limit of 10 inputs for both workflow_dispatch and workflow_call events is by design, not a bug. GitHub Actions enforces this limitation for the number of inputs defined in a workflow to prevent overly complex configurations, especially when used in the context of user-triggered workflows like workflow_dispatch.

However, while this makes sense for workflow_dispatch (where users manually trigger workflows from the UI), the same limit applies to workflow_call, which is used for calling reusable workflows from other workflows.
